gateway-service/src/test/java/org/zowe/apiml/gateway/security/service/HttpAuthenticationServiceTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.apiml.gateway.security.service;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.zowe.apiml.security.common.token.TokenAuthentication;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Optional;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class HttpAuthenticationServiceTest {
```

Comment:
I see that you work with JWT token. But there is also group of services that want basic auth and Gateway just forwards to the service. Did you consider acting on basic auth credentials as well?

Reply:
I am not sure we have the information. But if this is a case, of course it's something to be resolved.